### PR TITLE
Sanity check variantstore images before publishing [VS-889]

### PR DESCRIPTION
**Pull request by mcovarr**
tl;dr Added sanity checks, a smoke test, and fixed sample invocations in Docker build scripts.

Last month I built a bad base image when I was unsuccessfully trying to combine GCP and Azure CLIs in one image. What's worse is I seem to have tagged this broken image with a tag previously used for a good version of the image. These changes harden the Docker image build script to refuse to write over an existing tag and execute a smoke test against the image before pushing to GCR. While I was in there I also fixed the sample invocations that we saw during mobbing did not work on all versions of `date`.

### CODE REVIEW

File: scripts/variantstore/wdl/extract/build_build_base_docker.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 if [ $# -lt 1 ]; then
     echo "USAGE: ./build_build_base_docker.sh [DOCKER_TAG_STRING] [OPTIONAL:LATEST]"
-    echo " e.g.: ./build_build_base_docker.sh \$(date -I)-alpine-build-base"
+    echo " e.g.: ./build_build_base_docker.sh \$(date -Idate)-alpine-build-base"
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/wdl/extract/build_build_base_docker.sh:5]: Weird-looking but intentional. From the `date` man page on my machine:

```
     -I[FMT]
             Use ISO 8601 output format.  FMT may be omitted, in which case the default is date.  Valid FMT values are date, hours, minutes, and seconds.  The date and time is
             formatted to the specified precision.  When FMT is hours (or the more precise minutes or seconds), the ISO 8601 format includes the timezone.
```

I think what we saw with @RoriCremer last week was that she had a different version of `date` where the `FMT` was mandatory. It would be great to confirm that `date -Idate` does work on Rori's machine.

**Review by koncheto-broad — APPROVED** [review]
LGTM.  Happy to see a basic, sanity-checking smoke test happening inline right after building it

File: scripts/variantstore/wdl/extract/build_build_base_docker.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/build_build_base_docker.sh:5]: hahaha yes good idea
Screen Shot 2023-04-18 at 4 08 18 PM

File: scripts/variantstore/wdl/extract/build_build_base_docker.sh [hunk 2]
```diff
@@ -15,11 +15,35 @@ fi
...
 
+set +o errexit
+grep "${TAG}" <(docker images | grep "^${BASE_REPO}" | awk '{print $2}') > /dev/null
+RC="$?"
+set -o errexit
+
+if [[ $RC -eq 0 ]]
+then
+  echo "Error: Tag '${TAG}' already exists for image '${BASE_REPO}', refusing to tag again."
+  exit 1
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/extract/build_build_base_docker.sh:30]: smart--but also wont this get really annoying if we want to overwrite a docker image that we just made in error?

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/wdl/extract/build_build_base_docker.sh (hunk 2 above)
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/build_build_base_docker.sh:30]: ok ignore me, I just saw the next section about deleting it!
